libr/include/r_arch.h

```diff
@@ -0,0 +1,95 @@
+/* radare2 - LGPL - Copyright 2020 - pancake */
+
+#ifndef R2_ARCH_H
+#define R2_ARCH_H
+
+#include <r_util.h>
+#include <r_bind.h>
+
+typedef int RArchBits;
+typedef int RArchEndian;
+
+typedef enum r_arch_decode_options_t {
+	R_ARCH_OPTION_DISASM = 1,
+	R_ARCH_OPTION_SIZE = 2,
```

Comment:
can you provide examples of these as well in tests please? I'm asking because I'm not sure dealing with options inside each arch plugin is the best approach. For example, to get the size of an instruction you probably need to disassemble it anyway, so maybe it can be done once and for all in `r_arch_encode`/`r_arch_decode` instead of per-plugin. Similar for the other values in this enum probably, but some use cases as tests would help.

Reply:
i did one unit test, will add more over time